### ISSUE TITLE
"Race Condition" potential in setup flow

### ISSUE THREAD

**Issue by houllette** [erlef/elixir-secure-coding#19]
If the purposed architecture for autograding is to have the grader run on all MRs to main branch, it will block folks who are trying to make adjustments to the curriculum before having engineers in their company run it themselves.

So a break glass would need to be made in the grader job to allow for bypassing it (which is also useful for bugs in grading to not cause undo problems on participants), but we need to be careful not to introduce a hole so folks who aren't passing the test can just bypass it without completing the training (especially if the training is being used for compliance purposes).

**Comment by davydog187** [comment]
No longer relevant after #84
